tests/test_run_camera_playlist.py

```diff
@@ -0,0 +1,141 @@
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import unittest
+import sys
+import subprocess
+
+from lsst.ts import salobj
+from lsst.ts.IntegrationTests import ScriptQueueController
+from lsst.ts.IntegrationTests import RunCameraPlaylist
+from lsst.ts.IntegrationTests.configs.camera_playlist_configs import (
+    atcamera_playlists,
+    cccamera_playlists,
+)
+
+
+class RunCameraPlaylistTestCase(unittest.IsolatedAsyncioTestCase):
+    """Test the Run Camera Playlist integration test script."""
+
+    async def asyncSetUp(self):
+        # Set the LSST_DDS_PARTITION_PREFIX ENV_VAR.
+        salobj.set_random_lsst_dds_partition_prefix()
+
+        # Create the ScriptQueue Controller.
+        self.controller = ScriptQueueController(index=2)
+
+        # Start the controller and wait for it be ready.
+        await self.controller.start_task
+
+    async def test_run_camera_playlist(self):
+        """Execute the RunCameraPlaylist integration test script,
+        which runs the ts_standardscripts/run_command.py script.
+        Use the configuration stored in the camera_playlist_configs.py module.
+        """
+        # Mock the command-line arguments that the run_camera_playlist.py
+        # script expects.
+        test_camera = "at"
+        test_playlist = "test"
+        sys.argv[1:] = [test_camera, test_playlist]
+        # Instantiate the RunCameraPlaylist integration tests object and
+        # execute the scripts.
+        script_class = RunCameraPlaylist(
+            camera=test_camera, playlist_shortname=test_playlist
+        )
+        await script_class.run()
+        # Get number of scripts
+        num_scripts = len(script_class.scripts)
+        self.assertEqual(
+            script_class.playlist_config["parameters"]["playlist"],
+            atcamera_playlists[test_playlist],
+        )
+        print(
+            f"Running the {script_class.camera} "
+            f"{script_class.playlist_config['parameters']['playlist']}."
+        )
+        # Assert script was added to ScriptQueue.
+        self.assertEqual(len(self.controller.queue_list), num_scripts)
+
+    async def test_run_camera_playlist_bad_inputs(self):
+        """Attempt to execute the RunCameraPlaylist integration test script,
+        but use a bad set of command-line arguments; i.e. there is no
+        playlist for the given Camera.
+        """
+        # Mock the command-line arguments that the run_camera_playlist.py
+        # script expects.
+        test_camera = "cc"
+        test_playlist = "test"
+        # Instantiate the RunCameraPlaylist integration tests object and
+        # execute the scripts.
+        with self.assertRaises(Exception):
```

Comment:
You'll need to change this to look at `KeyError`.

Reply:
fixed